Flaky test: Publish terraform recipes failed with error: no matching resources found

Scheduled test failure:-

Run make publish-test-terraform-recipes
=> Creating Kubernetes namespace radius-test-tf-module-server...
kubectl create namespace radius-test-tf-module-server --dry-run=client -o yaml | kubectl apply -f -
namespace/radius-test-tf-module-server created
=> Publishing Terraform test recipes from ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes...
./.github/scripts/publish-test-terraform-recipes.py \
	./test/functional/shared/resources/testdata/recipes/test-terraform-recipes \
	radius-test-tf-module-server \
	tf-module-server-content
Publishing recipes from ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes to radius-test-tf-module-server/tf-module-server-content
Processing recipe: ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes/kubernetes-redis
Created zip file: /tmp/tmpi0py3tty/kubernetes-redis.zip
Processing recipe: ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes/wrong-output
Created zip file: /tmp/tmpi0py3tty/wrong-output.zip
Processing recipe: ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes/azure-storage
Created zip file: /tmp/tmpi0py3tty/azure-storage.zip
Processing recipe: ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes/parameter-outputs
Created zip file: /tmp/tmpi0py3tty/parameter-outputs.zip
Processing recipe: ./test/functional/shared/resources/testdata/recipes/test-terraform-recipes/k8ssecret-context
Created zip file: /tmp/tmpi0py3tty/k8ssecret-context.zip
configmap/tf-module-server-content created
Created configmap: radius-test-tf-module-server/tf-module-server-content
=> Deploying web server...
kubectl apply -f ./deploy/tf-module-server/resources.yaml -n radius-test-tf-module-server
deployment.apps/tf-module-server created
service/tf-module-server created
=> Waiting for web server to be ready...
kubectl wait --for=condition=ready pod -l  app.kubernetes.io/name=tf-module-server -n radius-test-tf-module-server --timeout=600s
error: no matching resources found
make: *** [build/recipes.mk:[46](https://github.com/radius-project/radius/actions/runs/7070240708/job/19246782044#step:24:47): publish-test-terraform-recipes] Error 1
Error: Process completed with exit code 2.


[AB#10653](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10653)

--- a/build/recipes.mk
+++ b/build/recipes.mk
@@ -47,6 +47,9 @@ publish-test-terraform-recipes: ## Publishes test terraform recipes to the curre
 	@echo "$(ARROW) Deploying web server..."
 	kubectl apply -f ./deploy/tf-module-server/resources.yaml -n $(TERRAFORM_MODULE_SERVER_NAMESPACE)
 
+	@echo "$(ARROW) Ensuring web server deployment is rolled out successfully..."
+	kubectl rollout status deployment.apps/tf-module-server -n $(TERRAFORM_MODULE_SERVER_NAMESPACE)
+
 	@echo "$(ARROW) Waiting for web server to be ready..."
 	kubectl wait --for=condition=ready pod -l  app.kubernetes.io/name=tf-module-server -n $(TERRAFORM_MODULE_SERVER_NAMESPACE) --timeout=600s
 

--- a/build/recipes.mk
+++ b/build/recipes.mk
@@ -47,11 +47,8 @@ publish-test-terraform-recipes: ## Publishes test terraform recipes to the curre
 	@echo "$(ARROW) Deploying web server..."
 	kubectl apply -f ./deploy/tf-module-server/resources.yaml -n $(TERRAFORM_MODULE_SERVER_NAMESPACE)
 
-	@echo "$(ARROW) Ensuring web server deployment is rolled out successfully..."
-	kubectl rollout status deployment.apps/tf-module-server -n $(TERRAFORM_MODULE_SERVER_NAMESPACE)
-
 	@echo "$(ARROW) Waiting for web server to be ready..."
-	kubectl wait --for=condition=ready pod -l  app.kubernetes.io/name=tf-module-server -n $(TERRAFORM_MODULE_SERVER_NAMESPACE) --timeout=600s
+	kubectl rollout status deployment.apps/tf-module-server -n $(TERRAFORM_MODULE_SERVER_NAMESPACE) --timeout=600s
 
 	@echo "$(ARROW) Web server ready. Recipes published to http://$(TERRAFORM_MODULE_SERVER_DEPLOYMENT_NAME).$(TERRAFORM_MODULE_SERVER_NAMESPACE).svc.cluster.local/<recipe_name>.zip"
 	@echo "$(ARROW) To test use:"

